### PR TITLE
ZkClient should close when the session is opened.

### PR DESCRIPTION
**Pull request by xjzrc**

### CODE REVIEW

File: kyuubi-main/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala [hunk 1]
```diff
@@ -84,28 +84,35 @@ class KyuubiSessionImpl(
...
-        val started = System.currentTimeMillis()
-        while (sh.isEmpty) {
-          if (process.waitFor(1, TimeUnit.SECONDS)) {
-            throw builder.getError
-          }
-          if (started + timeout <= System.currentTimeMillis()) {
-            process.destroyForcibly()
-            throw KyuubiSQLException(s"Timed out($timeout ms) to launched Spark with $builder",
-              builder.getError)
+    try {
```
Comment (yaooqinn) [inline kyuubi-main/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala:87]: Seems that we don't need this try-catch here as all are handled by the session manager.

How about just add this to the end?

```scala
try {
   zkClient.close()
} catch {
   case e: IOException => error("Failed to release the zkClient after session established", e)
}
```

Comment (xjzrc) [reply kyuubi-main/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala]: How about all kyuubi's sessions share one ZkClient?

Comment (yaooqinn) [reply kyuubi-main/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala]: I don‘t trust zk’s stability very much :)
For registering server or engines, we have a de-register handler there.

Comment (yaooqinn) [reply kyuubi-main/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala]: here, I don't think it is a right way to go

Comment (xjzrc) [reply kyuubi-main/src/main/scala/org/apache/kyuubi/session/KyuubiSessionImpl.scala]: i see

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
LGTM, pending CI

**Comment by yaooqinn** [comment]
thanks, merged to master

**Pull request closed without merging**